lib/utils/request.ts

```diff
@@ -0,0 +1,39 @@
+export type RequestFnType = (url: string, data: object) => void;
+
+export const requestWithSendBeacon: RequestFnType = (url, data) => {
+  const serializedData = JSON.stringify(data);
+  navigator.sendBeacon(url, serializedData);
+};
+
+export const requestWithXMLHttpRequest: RequestFnType = (url, data) => {
+  const serializedData = JSON.stringify(data);
+  const report = new XMLHttpRequest();
+  report.open("POST", url);
+  report.send(serializedData);
+};
+
+export const requestWithNodeHttpModule: RequestFnType = (url, data) => {
+  const serializedData = JSON.stringify(data);
+  const { protocol, host, path } = require("url").parse(url);
+  const options = {
+    protocol,
+    host,
+    path,
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "Content-Length": serializedData.length
+    }
+  };
+
+  const { request: nodeRequest } =
+    url.indexOf("https://") === 0 ? require("https") : require("http");
```

Comment:
if inlined in the other function, you can likely make this a static require

also I think `https` is always used, and the `https` module can deal with http too, but to check!

Reply:
The reason the require is there is, when it was a static require, the umd built file still contained 

```
  var httpsModule = require("https");
  var httpModule = require("http");
```

that code, while as it didn't include `requestWithNodeHttpModule` function.

And I will look into if `https` is covering `http`!

Reply:
Nope, `https` rejects requests with `http:` protocol.

Reply:
right now the insights endpoint works for both `http` and `https` but only `https` is documented.

> The Insights API endpoint can be reached via https://insights.algolia.io.
(source: https://www.algolia.com/doc/rest-api/insights/#overview)

So we could I guess only require the HTTPS module.
Let's leave it like this for now